mos-platform/atari2600-3e/mapper_3e.c

```diff
@@ -0,0 +1,73 @@
+// Copyright 2023 LLVM-MOS Project
+// Licensed under the Apache License, Version 2.0 with LLVM Exceptions.
+// See https://github.com/llvm-mos/llvm-mos-sdk/blob/main/LICENSE for license
+// information.
+
+#include <peekpoke.h>
+#include <mapper.h>
+
+__attribute__((section(".zp.bss"))) bank_index_t __current_bank;
+
+__attribute__((weak)) void set_current_bank(bank_index_t bank_id) {
+  if (bank_select && !(bank_id & 0x80)) {
```

Comment:
This is checking whether the address of bank_select is null, but it's defined in the same file, so it could never be. Is something else intended here?

Reply:
That was due to my misunderstanding of the weak attribute, I thought it pruned functions that are never called by non-weak functions. But it turns out the optimizer figures out that bank_id is never set >= 128 and can optimize it anyway :) Anyway I removed all the weak attributes.